vnext/Microsoft.ReactNative/TurboModulesProvider.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include <TurboModuleRegistry.h>
+
+namespace winrt::Microsoft::ReactNative {
+
+class TurboModulesProvider final : public facebook::react::TurboModuleRegistry {
+ public:
+  virtual std::shared_ptr<facebook::react::TurboModule> getModule(
+      const std::string &moduleName,
+      const std::shared_ptr<facebook::react::CallInvoker> &callInvoker) noexcept override;
+  virtual std::vector<std::string> getEagerInitModuleNames() noexcept override;
+
+ public:
+  void AddModuleProvider(winrt::hstring const &moduleName, ReactModuleProvider const &moduleProvider) noexcept;
+
+ private:
+  std::map<std::string, ReactModuleProvider> m_moduleProviders;
```

Comment:
avoid use of std::map

Reply:
Is there any better choice? I need exactly a map here.

Reply:
@ZihanChen-MSFT a few options:
* `std::vector<std::pair<string, ReactModuleProvider>>`
* `f14map` (https://engineering.fb.com/developer-tools/f14/)
* `std::unordered_map` although if you expect to have less than 1000 entries, don't use a map

Reply:
I will use unordered_map for now, I don't quite believe we will have more than 1000 turbo modules running in an app.